skbio/sequence/_interval.py

```diff
@@ -0,0 +1,133 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+from __future__ import absolute_import, division, print_function
+
+from ._feature import Feature
+from .intersection import Interval, IntervalTree
+import itertools
+from collections import defaultdict
+
+class IntervalMetadataMixin():
+    ''' Store metadata corresponding to Features and Intervals.
+
+    There are two underlying data structures namely
+    1. An IntervalTree (interval_metadata)
+    2. A hash table indexed by a key and value (feature_table)
+
+    Attributes
+    ----------
+    interval_metadata
+    feature_table
+
+    '''
+    def __init__(self, interval_metadata=None, feature_table=None):
+        self._init_(self, interval_metadata=None, feature_table=None)
+
+    def _init_(self, interval_metadata=None, feature_metadata=None):
```

Comment:
I based this on PositionalMetadataMixin

https://github.com/biocore/scikit-bio/blob/master/skbio/_base.py#L336



On Mon, Feb 8, 2016 at 1:53 PM, Zech Xu <notifications@github.com> wrote:

> In skbio/sequence/_interval.py
> <https://github.com/RNAer/scikit-bio/pull/1#discussion_r52227737>:
>
> > +    ''' Store metadata corresponding to Features and Intervals.
> > +
> > +    There are two underlying data structures namely
> > +    1. An IntervalTree (interval_metadata)
> > +    2. A hash table indexed by a key and value (feature_table)
> > +
> > +    Attributes
> > +    ----------
> > +    interval_metadata
> > +    feature_table
> > +
> > +    '''
> > +    def __init__(self, interval_metadata=None, feature_table=None):
> > +        self._init_(self, interval_metadata=None, feature_table=None)
> > +
> > +    def _init_(self, interval_metadata=None, feature_metadata=None):
>
> why not just put it in __init__?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RNAer/scikit-bio/pull/1/files#r52227737>.
>